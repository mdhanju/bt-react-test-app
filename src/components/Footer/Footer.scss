footer {
    &#btrc-Footer {
        padding-bottom: 1.4rem;
        font-size: 1.5rem;
        color: $dark-indigo;
        @extend .hca-book;
    }
}
