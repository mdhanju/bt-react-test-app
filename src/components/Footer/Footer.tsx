import React from 'react'
import classnames from 'classnames'
import { FormattedMessage } from 'react-intl'

const Footer = ({ className, id, ...props }: Props): React.ReactElement => (
    <footer
        id={id}
        className={classnames(className, 'container-fluid')}
        {...props}
    >
        &copy; {new Date().getFullYear()}&nbsp;
        <FormattedMessage
            id="app.footername"
            description="Footer name showing Blue Triangle name"
        />
    </footer>
)

type Props = {
    id?: string;
    className?: string;
};

Footer.defaultProps = {
    id: 'btrc-Footer'
}

export default Footer
