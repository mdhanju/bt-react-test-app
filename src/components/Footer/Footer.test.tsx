/* eslint-disable react/jsx-no-undef */

import React from 'react'

import Footer from './Footer'

describe('<Footer />', () => {
    it('should render initial layout', () => {
        const wrapper = shallow(<Footer />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('should shallow render message', () => {
        const expectedInnerText = `© ${new Date().getFullYear()}\u00a0<FormattedMessage />`
        const wrapper = shallow(<Footer />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.text()).toBe(expectedInnerText)
    })

    it('render the localized (En) footer', () => {
        const wrapper = render(
            <Intl locale="en">
                <Footer />
            </Intl>
        )
        expect(wrapper.text()).toBe(
            `© ${new Date().getFullYear()}\u00a0Blue Triangle, Inc.`
        )
    })
})
