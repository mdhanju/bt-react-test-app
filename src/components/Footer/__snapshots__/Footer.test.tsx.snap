// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render initial layout 1`] = `
Array [
  <footer
    className="container-fluid"
    id="btrc-Footer"
  >
    © 
    2020
     
    <FormattedMessage
      description="Footer name showing Blue Triangle name"
      id="app.footername"
      values={Object {}}
    />
  </footer>,
]
`;
